### ISSUE TITLE
Allow the ability to set shareNativeConnection to false for LettuceConnectionFactory

### ISSUE THREAD

**Issue by csavory** [spring-projects/spring-boot#14196]
By default, the `LettuceConnection` will use the `sharedConnection` for almost all non-tx or non-blocking operations. 

I have found the exclusive use of the shared connection to not allow very high throughput of many concurrent Redis calls. My reads connections are constantly being blocked by `LettuceFutures.awaitOrCancel`.  Therefore I would like to set `shareNativeConnection` to false in the `LettuceConnectionFactory` so it always uses the pooled connections.  

Spring Boot `LettuceConnectionConfiguration` does not currently allow for `shareNativeConnection` to be set.

https://docs.spring.io/spring-data/redis/docs/2.1.0.RC2/reference/html/#redis:connectors:lettuce

**Comment by philwebb** [comment]
@csavory For now you'll need to define your own `LettuceConnectionFactory` `@Bean`.

**Comment by philwebb** [comment]
Closing in favor of PR #14217

**Comment by mp911de** [comment]
We should clarify one thing first before extending configuration: Redis itself is single-threaded so increasing the connection count for Redis Standalone does not improve throughput performance. 

@csavory if you recognize a single connection as an obstruction for your application, then we should continue the discussion along with some profiling data in https://github.com/lettuce-io/lettuce-core.

**Comment by snicoll** [comment]
@csavory switching to declined based on the comment above. Please follow up on the lettuce project and we can reopen if it turns out something needs to be done in Spring Boot.
